structcol/montecarlo.py

```diff
@@ -328,82 +431,141 @@ def RTcounter(z, z_low, cutoff, ntraj, n_matrix, n_sample, kx, ky, kz):
+        # is phi1 = phi3, so append their average:
         if B == 0:
             phi_r.append((phi1+phi3)/2)
         elif B == 1:
             phi_r.append((phi1+phi4)/2)
         elif B == 2:
             phi_r.append((phi2+phi3)/2)
         elif B == 3:
-            phi_r.append((2*np.pi+phi2+phi4)/2)
-
-        theta_r.append(theta)
+            phi_r.append((phi2+phi4)/2)
 
-        # Count how many of the thetas correspond to the range of total internal reflection
+        # Count how many of the thetas correspond to the range of total 
+        # internal reflection
         if theta < theta_min_refracted:
             count = count + 1
 
+        # (Keep until further revisions):
+        #phi2 = -np.arccos(cosA[i] / np.sin(theta))
+        #phi3 = np.arcsin(cosB[i] / np.sin(theta))
+        #phi4 = np.pi - np.arcsin(cosB[i] / np.sin(theta))
+        #A = np.array([abs(phi1-phi3),abs(phi1-phi4),abs(phi2-phi3),abs(2*np.pi+phi2-phi4)])
+        #elif B == 3:
+        #    phi_r.append((2*np.pi+phi2+phi4)/2)
 
     # Calculate corrected reflection fraction
-
     R_fraction_corrected = np.array(len(R_row_indices) - count) / ntraj
 
-
     return R_fraction_corrected, theta_r, phi_r
 
 
 def initialize(nevents, ntraj, seed=None):
     """
-    Sets the trajectory's initial conditions (position, direction, and weight).
-
-    nevents: number of scattering events
-    ntraj: number of trajectories
-
-    TODO: add docstring in numpy docstring format
+    Sets the trajectories' initial conditions (position, direction, and weight).
+    
+    The initial positions are determined randomly in the x-y plane (the initial 
+    z-position is at z = 0). The initial propagation direction is set to be 1 
+    at z, meaning that the photon packets point straight down in z. The initial 
+    weight is currently determined to be a value of choice. 
+
+    Parameters
+    ----------  
+    nevents : int
+        Number of scattering events
+    ntraj : int
+        Number of trajectories
+    seed : int or None        
+        If seed is int, the simulation results will be reproducible. If seed is
+        None, the simulation results are actually random. 
+       
+    Returns 
+    ---------- 
+    r0 : array_like (structcol.Quantity [length])
+        Initial position.
+    k0 : array_like (structcol.Quantity [dimensionless])
+        Initial direction of propagation.
+    weight0 : array_like (structcol.Quantity [dimensionless])
+        Initial weight.
+    
     """
-
+        
     if seed is not None:
         np.random.seed([seed])
 
-    # initial position
+    # Initial position. The position array has one more row than the direction
+    # and weight arrays because in includes the starting positions on the x-y 
+    # plane
     r0 = np.zeros((3, nevents+1, ntraj))
     r0[0,0,:] = random((1,ntraj))
     r0[1,0,:] = random((1,ntraj))
 
-    # initial direction
+    # Initial direction
     eps = 1.e-9
```

Comment:
avoid hardcoding constants.  These should be specified as variables in the function, with default values that can be overridden.